modules/renter/proto/negotiate.go

```diff
@@ -172,26 +184,35 @@ func negotiateRevision(conn net.Conn, rev types.FileContractRevision, secretKey
 func newRevision(current types.FileContractRevision, cost types.Currency) types.FileContractRevision {
 	rev := current
 
-	// need to manually copy slice memory
-	rev.NewValidProofOutputs = make([]types.SiacoinOutput, 2)
-	rev.NewMissedProofOutputs = make([]types.SiacoinOutput, 3)
-	copy(rev.NewValidProofOutputs, current.NewValidProofOutputs)
-	copy(rev.NewMissedProofOutputs, current.NewMissedProofOutputs)
-
-	// move valid payout from renter to host
-	rev.NewValidProofOutputs[0].Value = current.NewValidProofOutputs[0].Value.Sub(cost)
-	rev.NewValidProofOutputs[1].Value = current.NewValidProofOutputs[1].Value.Add(cost)
+	rev.NewValidProofOutputs = newPayoutAdjustment(payoutAdjustmentInput{
+		outputs:    current.NewValidProofOutputs,
+		payerIndex: types.FileContractRenterIndex,
+		payeeIndex: types.FileContractHostIndex,
+		amountPaid: cost})
```

Comment:
rather than a new struct type, I think I'd prefer a helper function:
```go
func moveOutputValue(outputs []types.SiacoinOutput, amount types.Currency, from, to int) []types.SiacoinOutput {
    newOutputs := append([]types.SiacoinOutput(nil), outputs...)
    newOutputs[from].Value = newOutputs[from].Value.Sub(amount)
    newOutputs[to].Value = newOutputs[to].Value.Add(amount)
    return newOutputs
}

func newRevision(current types.FileContractRevision, cost types.Currency) types.FileContractRevision {
    rev := current
    // move valid payout from renter to host
    rev.NewValidProofOutputs = moveOutputValue(
        rev.NewValidProofOutputs,
        cost,
        modules.FileContractRenterIndex,
        modules.FileContractHostIndex,
    )
    // move missed payout from renter to void
    rev.NewMissedProofOutputs = moveOutputValue(
        rev.NewMissedProofOutputs,
        cost,
        modules.FileContractRenterIndex,
        modules.FileContractVoidIndex,
    )
    // increment revision number
    rev.NewRevisionNumber++
    return rev
}
```

Reply:
Feels like a lot of different variations for a basic cleanup task. I was partial to the one David led me to before and also like the one @avahowell helped me realize with https://github.com/NebulousLabs/Sia/pull/2353  Mostly I wanted to do this task to try to get "something" improved in terms of clarity in less than a few hours.

Reply:
I definitely prefer the style @lukechampine just presented, it's readable and doesn't introduce a whole new struct. Switching out the `from` and `two` ints to named consts really improves the understanding for what's happening here.

This pattern appears all over the `proto` package, if we're going to clean stuff up, it makes sense to do it right. When you put a lot of effort into getting the first few steps correct, the rest are a lot easier and you end up with a final product that you are much happier with.